### PR TITLE
[rag eval] make less sheets reads, fix more table parsing

### PR DESCRIPTION
**Pull request by chejennifer**
- make one call to read mulitple rows instead of making one call per row
- fix table parsing when header contains "," or "₂"

### CODE REVIEW

File: static/js/apps/eval_retrieval_generation/data_store.tsx [hunk 1]
```diff
@@ -187,3 +192,32 @@ export async function getCallCount(
...
+ */
+export async function getSheetsRows(
+  sheet: GoogleSpreadsheetWorksheet,
+  rowIdxList: number[]
+): Promise<Record<number, GoogleSpreadsheetRow>> {
+  if (_.isEmpty(rowIdxList)) {
+    return Promise.resolve({});
+  }
+  rowIdxList.sort((a, b) => a - b);
+  const firstRowIdx = rowIdxList[0];
```
Comment (shifucun) [inline static/js/apps/eval_retrieval_generation/data_store.tsx:210]: also have lastRowIdx = ...?

File: static/js/apps/eval_retrieval_generation/util.ts [hunk 2]
```diff
@@ -24,7 +24,7 @@ const LONG_SPACES = "&nbsp;&nbsp;&nbsp;&nbsp;";
 const TABLE_DIVIDER_PATTERN = /[--][-]+/g;
 // table headers are sometimes country names so there can be symbols used like
 // in "Côte d'Ivoire".
-const TABLE_HEADER_TEXT_PATTERN = /[\w'ô[\]ãéí\s°()%:-]+/g;
+const TABLE_HEADER_TEXT_PATTERN = /[\w'ô[\]ãéí\s°()%:-\\,\\₂]+/g;
```
Comment (shifucun) [inline static/js/apps/eval_retrieval_generation/util.ts:27]: this is a mystery now :)

File: static/js/apps/eval_retrieval_generation/data_store.tsx [hunk 3]
```diff
@@ -187,3 +192,32 @@ export async function getCallCount(
...
   return snapshot.data().count;
 }
+
+
+/**
+ * Returns a promise to get a record of row idx to google sheet row
+ * @param sheet the sheet to get rows from
+ * @param rowIdxList the list of rows to get from the sheet
+ */
+export async function getSheetsRows(
```
Comment (shifucun) [inline static/js/apps/eval_retrieval_generation/data_store.tsx:202]: very nice!!!

File: static/js/apps/eval_retrieval_generation/data_store.tsx (hunk 1 above)
Comment (chejennifer) [reply static/js/apps/eval_retrieval_generation/data_store.tsx:209]: done

File: static/js/apps/eval_retrieval_generation/util.ts (hunk 2 above)
Comment (chejennifer) [reply static/js/apps/eval_retrieval_generation/util.ts:27]: it's essentially every character that can be in the header

**Review by shifucun — APPROVED** [review]

**Comment by chejennifer** [comment]
thanks for the review!

**Comment by pradh** [comment]
Nice!!  Thanks for this PR!

**Pull request merged**